ZuluContent/Engines/Harvest/Core/HarvestSystem.cs

```diff
@@ -1,504 +1,500 @@
+            return CheckTool(from, tool);
+        }
+
+        public virtual bool CheckHarvest(Mobile from, Item tool, HarvestDefinition def, object toHarvest)
+        {
+            return CheckTool(from, tool);
+        }
+
+        public virtual bool CheckRange(Mobile from, Item tool, HarvestDefinition def, Map map, Point3D loc, bool timed)
+        {
+            bool inRange = @from.Map == map && @from.InRange(loc, def.MaxRange);
+
+            if (!inRange)
+                def.SendMessageTo(from, timed ? def.TimedOutOfRangeMessage : def.OutOfRangeMessage);
+
+            return inRange;
+        }
+
+        public virtual bool CheckResources(Mobile from, Item tool, HarvestDefinition def, Map map, Point3D loc,
+            bool timed)
+        {
+            HarvestBank bank = def.GetBank(map, loc.X, loc.Y);
+            bool available = bank != null && bank.Current >= GetConsumeAmount(from, def);
+
+            if (!available)
+                def.SendMessageTo(from, timed ? def.DoubleHarvestMessage : def.NoResourcesMessage);
+
+            return available;
+        }
+
+        public virtual int GetConsumeAmount(Mobile from, HarvestDefinition def)
+        {
+            var skillValue = from.Skills[def.Skill].Value;
+            return def.ConsumedPerHarvest(skillValue);
+        }
+
+        public virtual int GetChanceForColored(Mobile from, Item tool, HarvestDefinition def)
+        {
+            var skillValue = from.Skills[def.Skill].Value;
+            var chanceForColored = (int) (skillValue / 5) + 35;
+
+            if (tool is IEnchanted enchantedTool && enchantedTool.Enchantments.Get((HarvestBonus e) => e.Value) > 0)
+            {
+                var toolBonusChanceForColored = 10;
+                enchantedTool.FireHook(h => h.OnHarvestBonus(from, ref toolBonusChanceForColored));
+                chanceForColored += toolBonusChanceForColored;
+            }
+
+            if (chanceForColored > 75)
+                chanceForColored = 75;
+
+            from.FireHook(h => h.OnHarvestColoredChance(from, ref chanceForColored));
+
+            return chanceForColored;
+        }
+
+        public virtual void OnBadHarvestTarget(Mobile from, Item tool, object toHarvest)
+        {
+        }
+
+        public virtual object GetLock(Mobile from, Item tool, HarvestDefinition def, object toHarvest)
+        {
+            /* Here we prevent multiple harvesting.
+             * 
+             * Some options:
+             *  - 'return tool;' : This will allow the player to harvest more than once concurrently, but only if they use multiple tools. This seems to be as OSI.
+             *  - 'return GetType();' : This will disallow multiple harvesting of the same type. That is, we couldn't mine more than once concurrently, but we could be both mining and lumberjacking.
+             *  - 'return typeof( HarvestSystem );' : This will completely restrict concurrent harvesting.
+             */
+
+            return tool;
```

Comment:
Judging from the comments this should be `return typeof( HarvestSystem );` to match ZH?

Reply:
Cool done